### DISCUSSION TITLE
Почему Playlist.saveWithUpdate добавляет треки в начало плейлиста?

### DISCUSSION THREAD

**Discussion by efremych** [Chimildic/goofy#71] [Помощь с ошибкой] [answered]
Не пойму, почему saveWithUpdate добавляет треки в начало плейлиста?

судя по этой строчке должно добавляться в конец, но это не так
`                addTracks({ id: data.id, tracks: tracksToAdd, toEnd: data.toEnd });`

по аналогии с saveWithAppend изменил строчку в лоб на так и теперь норм
`addTracks({ id: data.id, tracks: tracksToAdd, toEnd: true});`

Что я не так понимаю?

**Comment by Chimildic** [comment, marked as answer]
Читаем [документацию](https://chimildic.github.io/goofy/#/func?id=savewithupdate), есть описание _дополнительного_ режима, связанного с `data.toEnd`. Аналогично для `saveWithAppend`.
Если не указывать `toEnd` явно при вызове функции, по умолчанию добавление в начало. И наоборот, указав со значением `true`, добавление в конец.
```js
Playlist.saveWithUpdate({
    name: 'Имя',
    tracks: tracks,
    toEnd: true,
});
```

В случае следующей строчки: `toEnd: data.toEnd` - означает подстановку булева значения, переменной, а не явных инструкций по добавления в конец. Для `JavaScript` отсутствие значения (не указали при вызове функции) приравнивается к `false`. Поэтому добавляется в начало. 
```js
addTracks({ id: data.id, tracks: tracksToAdd, toEnd: data.toEnd });
```

Ваша правка ломает механизм. При любом вызове будет добавление в конец.
```js
addTracks({ id: data.id, tracks: tracksToAdd, toEnd: true});
```

**Comment by efremych** [comment]
```
Playlist.saveWithUpdate({
    name: 'Имя',
    tracks: tracks,
    toEnd: true,
});
```

Да, спасибо, мой косяк - не для того плейлиста вставлял `toEnd: true,`, когда проверял возможные варианты.
